website/static/js/nodesDelete.js

```diff
@@ -1,114 +1,138 @@
 /**
- * Controller for deleting a node and it childs (if they exists)
+ * Controller for deleting a node and its children (if they exist)
  */
 'use strict';
 
 var $ = require('jquery');
 var ko = require('knockout');
 var Raven = require('raven-js');
 var $osf = require('./osfHelpers');
-var osfHelpers = require('js/osfHelpers');
 var m = require('mithril');
 var bootbox = require('bootbox');
 var NodesDeleteTreebeard = require('js/nodesDeleteTreebeard');
 
-function _flattenNodeTree(nodeTree) {
-    var ret = [];
-    var stack = [nodeTree];
-    while (stack.length) {
-        var node = stack.pop();
-        ret.push(node);
-        stack = stack.concat(node.children);
-    }
-    return ret;
-}
 
-/**
- * take treebeard tree structure of nodes and get a dictionary of parent node and all its
- * children
- */
-function getNodesOriginal(nodeTree, nodesOriginal) {
-    var flatNodes = _flattenNodeTree(nodeTree);
-    $.each(flatNodes, function(_, nodeMeta) {
-        nodesOriginal[nodeMeta.node.id] = {
-            id: nodeMeta.node.id,
-            title: nodeMeta.node.title,
-            isAdmin: nodeMeta.node.is_admin,
-            changed: false
-        };
-    });
-    nodesOriginal[nodeTree.node.id].isRoot = true;
-    return nodesOriginal;
-}
+var BULK_DELETE_LIMIT = 100;
 
 /**
- * Deletes all nodes in changed state
- * uses API v2 bulk requests
+ * view model which corresponds to nodes_delete.mako (#nodesDelete)
+ * Used for quickly deleting nodes with no children.
+ * @type {QuickDeleteViewModel}
  */
-function batchNodesDelete(nodes) {
-    var nodesV2Url = window.contextVars.apiV2Prefix + 'nodes/';
-    var nodesBatch = $.map(nodes, function (node) {
-        return {
-            'type': 'nodes',
-            'id': node.id,
-        };
+
+var QuickDeleteViewModel = function (nodeType, isPreprint, nodeApiUrl) {
+    var self = this;
+
+    self.confirmationString = $osf.getConfirmationString();
+    self.QUICKDELETE = 'quickDelete';
+    self.SELECT = 'select';
+    self.CONFIRM = 'confirm';
+    self.page = ko.observable(self.QUICKDELETE);
+    self.confirmInput = ko.observable('');
+    self.canDelete = ko.observable(false);
+    self.nodeType = nodeType;
+    self.nodeApiUrl = nodeApiUrl;
+    self.isPreprint = isPreprint;
+    self.preprintMessage = '<br><br>This ' + self.nodeType +
+      ' also contains a <strong>preprint</strong>. Deleting this ' + self.nodeType +
+      ' will also delete your <strong>preprint</strong>. This action is irreversible.';
+
+    self.message = ko.computed(function () {
+        return 'It will no longer be available to other contributors on the project.' +
```

Comment:
Does this portion of the message need to be in the NodesDeleteViewModel as well?